### PR TITLE
feat(wip): scheduled delay feature

### PR DESCRIPTION
**Pull request by ainouzgali**
- **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Feature

- **Why this change was needed?** (You can also link to an open issue here)
To let user input a date that after which, the following steps will be triggered

- **Other information**:
Screen Shot 2022-09-22 at 20 11 41

Screen Shot 2022-09-21 at 18 15 50
Screen Shot 2022-09-21 at 18 17 36

### CODE REVIEW

File: apps/api/src/app/events/services/workflow.queue.service.ts [hunk 1]
```diff
@@ -173,22 +174,26 @@ export class WorkflowQueueService {
...
+    const timeNow = new Date();
     await this.jobRepository.updateStatus(data._id, JobStatusEnum.DELAYED);
 
     let delay: number;
-    if (WorkflowQueueService.checkValidDelayOverride(data)) {
-      delay = WorkflowQueueService.toMilliseconds(
-        data.overrides.delay.amount as number,
-        data.overrides.delay.unit as DigestUnitEnum
-      );
+    if (data.step.metadata.type === DelayTypeEnum.SCHEDULED) {
```
Comment (davidsoderberg) [inline apps/api/src/app/events/services/workflow.queue.service.ts:186]: maybe work extracting the delay calculation to a method now?

File: apps/api/src/app/events/usecases/verify-payload/verify-payload.usecase.ts [hunk 2]
```diff
@@ -60,6 +66,22 @@ export class VerifyPayload {
     return invalidKeys;
   }
 
+  private checkRequiredDelayPath(delayPath: string, payload: Record<string, unknown>): string {
+    const invalidKey = `${delayPath} (ISO Date)`;
+
+    if (!payload.hasOwnProperty(delayPath)) {
+      return invalidKey;
+    } else {
```
Comment (davidsoderberg) [inline apps/api/src/app/events/usecases/verify-payload/verify-payload.usecase.ts:74]: else not needed here?

Comment (ainouzgali) [reply apps/api/src/app/events/usecases/verify-payload/verify-payload.usecase.ts:74]: True. forgot to remove it

File: apps/api/src/app/events/services/workflow.queue.service.ts [hunk 3]
```diff
@@ -206,6 +202,29 @@ export class WorkflowQueueService {
     );
   }
 
+  private static calculateDelayAmount(data: JobEntity): number {
```
Comment (davidsoderberg) [inline apps/api/src/app/events/services/workflow.queue.service.ts:205]: This is awesome :)

**Review by davidsoderberg — APPROVED** [review]
🥇 Looks great 😄

**Pull request merged**